### PR TITLE
[PATCH v6] DPDK detection through pkg-config

### PR DESCRIPTION
**Pull request by lumag**

### CODE REVIEW

**Comment by lumag** [comment]
Passed now

**Comment by muvarov** [comment]
I'm ok with all 3 patches.

File: m4/odp_dpdk.m4 [hunk 1]
```diff
@@ -173,6 +173,15 @@ AC_DEFUN([_ODP_DPDK_LEGACY], [dnl
     AC_SUBST([DPDK_PKG])
 ])
 
+m4_ifndef([PKG_CHECK_MODULES_STATIC],
```
Comment (Bill-Fischofer-Linaro) [inline m4/odp_dpdk.m4]: Typo in this file's commit msg. Should be "let's provide it on our own".

File: m4/odp_dpdk.m4 [hunk 2]
```diff
@@ -143,21 +147,58 @@ AS_IF([test "x$1" = "xsystem"], [dnl
...
+# Check for DPDK availability
+AC_DEFUN([ODP_DPDK], [dnl
+AS_IF([test "x$1" = "xsystem"],
+      [PKG_CHECK_MODULES([DPDK], [libdpdk],
+			 [AC_MSG_NOTICE([Using DPDK detected via pkg-config])
+			 _ODP_DPDK_PKGCONFIG
+			 m4_default([$2], [:])],
+			 [_ODP_DPDK_LEGACY_SYSTEM([m4_default([$2], [:])],
+						  [m4_default([$3], [:])])])],
+      [_ODP_DPDK_LEGACY($1, [m4_default([$3], [:])], [m4_default([$4], [:])])]
```
Comment (matiaselo) [inline m4/odp_dpdk.m4]: Shouldn't this line be?
`      [_ODP_DPDK_LEGACY($1, [m4_default([$2], [:])], [m4_default([$3], [:])])]`

Comment (lumag) [reply m4/odp_dpdk.m4]: Hmm, looks so. Does it work this way?

Comment (matiaselo) [reply m4/odp_dpdk.m4]: Yes, it fixes the configure problem I was having.

**Comment by lumag** [comment]
Updated

**Comment by Bill-Fischofer-Linaro** [comment]
Looks like the latest version is still seeing a Travis glitch:
```
echo "TESTS=\"validation/api/pktio/pktio_run.sh validation/api/pktio/pktio_run_tap.sh validation/api/shmem/shmem_linux validation/api/pktio/pktio_run_pcap.sh mmap_vlan_ins/mmap_vlan_ins.sh  validation/api/pktio/pktio_run_dpdk.sh pktio_ipc/pktio_ipc_run.sh\""    > tests-linux-generic.env
echo "ODP_PLATFORM=linux-generic EXEEXT= TEST_DIR=../../../test/validation" >> tests-linux-generic.env
echo ""      >> tests-linux-generic.env
make  check-TESTS
make[3]: Entering directory '/odp/platform/linux-generic/test'
make[4]: Entering directory '/odp/platform/linux-generic/test'
PASS: validation/api/pktio/pktio_run.sh
PASS: validation/api/pktio/pktio_run_tap.sh
PASS: validation/api/shmem/shmem_linux
PASS: validation/api/pktio/pktio_run_pcap.sh
PASS: mmap_vlan_ins/mmap_vlan_ins.sh
FAIL: validation/api/pktio/pktio_run_dpdk.sh
PASS: pktio_ipc/pktio_ipc_run.sh
```

**Comment by matiaselo** [comment]
> Looks like the latest version is still seeing a Travis glitch:
> 
> ```
> echo "TESTS=\"validation/api/pktio/pktio_run.sh validation/api/pktio/pktio_run_tap.sh validation/api/shmem/shmem_linux validation/api/pktio/pktio_run_pcap.sh mmap_vlan_ins/mmap_vlan_ins.sh  validation/api/pktio/pktio_run_dpdk.sh pktio_ipc/pktio_ipc_run.sh\""    > tests-linux-generic.env
> echo "ODP_PLATFORM=linux-generic EXEEXT= TEST_DIR=../../../test/validation" >> tests-linux-generic.env
> echo ""      >> tests-linux-generic.env
> make  check-TESTS
> ...
> PASS: mmap_vlan_ins/mmap_vlan_ins.sh
> FAIL: validation/api/pktio/pktio_run_dpdk.sh
> PASS: pktio_ipc/pktio_ipc_run.sh
> ```

This seems unrelated to this PR. It looks like the VM CPU is not supporting required DPDK feature (RDRAND). Could you please restart the test.

**Comment by Bill-Fischofer-Linaro** [comment]
This PR is still having Travis issues. Not sure what's needed to resolve them but restarting the failing test doesn't help.

**Comment by muvarov** [comment]
Merged,

**Pull request closed without merging**